pkg/core/object/metadata.go

```diff
@@ -0,0 +1,115 @@
+package object
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"slices"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+)
+
+// SearchResultMerger merges SearchV2 results from several sources. Must not be
+// used concurrently.
+type SearchResultMerger struct {
+	limit, n    uint16
+	more        bool
+	buf         []client.SearchResultItem
+	iBuf, iNext *big.Int
+}
+
+// NewSearchResultMerger constructs SearchResultMerger merging up to limit
+// results.
+func NewSearchResultMerger(limit uint16) *SearchResultMerger {
+	return &SearchResultMerger{limit: limit}
+}
+
+// Add accumulates the next result.
+func (x *SearchResultMerger) Add(withCursor bool, next []client.SearchResultItem) {
```

Comment:
This is not very efficient, you're iterating over the whole result set again and again replacing some elements. Operating over the whole set of results would be much more efficient, you just take whatever is the minimal one at each step until you have enough results. This requires a bit more memory, but given our limits it's ok.

Reply:
done